apps/silverback-website/docs/drupal/modules.mdx

```diff
@@ -0,0 +1,72 @@
+---
+title: Contributed modules
+path: /drupal/modules
+nav: Drupal
+position: 0
+---
+
+# Contributed Drupal modules
+
+This document is a comprehensive list of Drupal extensions that we support to be
+used in projects. Other modules _must not_ be installed on a per-project basis.
+If Drupal Core in combination with extensions on this list is not enough to meet
+specific requirements, there are three possible paths:
+
+1. Implement a custom, in-project module with fully automated test coverage.
+2. Add a pull request to this document, adding the desired module.
+3. Contribute to any module on this list to add the required features.
+
+Adding arbitrary modules with varying quality to projects whenever a requirement
+arises should be avoided. As well as patching modules and thereby creating
+unstable dependencies.
+
+
+## Confirmed modules
+
+### GraphQL version 4
+
+GraphQL is the default solution for exposing data and operations to the
+frontend. Every project should have its own custom GraphQL schema that documents
+the domain model and acts as a contract been Drupal and all consumers.
+For in-depth on how to create custom GraphQL schemas, please refer to the
+[module documentation](https://drupal-graphql.gitbook.io/graphql/).
+
+### Admin Toolbar
+
+### Pathauto
+
+### Redirect
+
+### Webform
+
+### Gatsby
+
+### Token
+
+### Field group
+
+### Linkit
+
+### Search API
+
+### Field permissions
+
+### Address field
+
+### Focal Point
+
+## Prohibited modules
+Don't even bother to start a discussion.
+
+### Paragraphs
+Paragraphs generate immense load on the database and lead to lacking (up to non-
+existent) content modeling. Compared to other page building solutions, the
+editing experience is abysmal, it produces a high level of confusion and
+complexity in multilingual setups and should be avoided at all cost.
+
+### JSON:API
```

Comment:
I know I'm not supposed to start a discussion here but... sometimes json-api might be more than enough to cover the client or project needs, so banning beforehand feels too radical.

Reply:
This was exactly my opinion, but the last months working with changed that. If we consume the API ourselves (Gatsby or other frontends), we hurt us by not having the flexibility of GraphQL, and when creating a public API, exposing all the Drupalisms is not a good idea anyway.

Recent experience has shown that GraphQL is by the cheaper option, and no matter the Budget, each project should spend the least possible amount of money.

Reply:
@fjgarlin is it an often case when you need to expose a lot (or almost everything) of Drupal's data?

My experience with GraphQL v4 on Gatsby-Drupal prototype:
- 1.5 hours to learn how to write resolvers and custom data producers.
- About 6 hours in total to implement the whole schema and update the frontend code (GraphQL v3 was used before).

Of course the [schema was small](https://github.com/AmazeeLabs/silverback-mono/blob/f4edd625cac4f49f0df53a45e1616505d28a506f/apps/silverback-drupal/web/modules/custom/silverback_gatsby/graphql/silverback_gatsby.graphqls), but I spent not that much of time implementing it and the frontend DX was significantly improved.

Reply:
Thinking on the cases where we don't need to expose a lot of data, like a small extension where part of the site is exposed.

My point is that, whilst it might not be a suitable solution for most cases, it might still be a good fit for some, where the requirements or integrations aren't too complex.

In any case, this is more maintenance point of view (ie: extension) rather than building a project from scratch, so all your points above are still valid I guess and it's no big deal to leave it out from the `silverback-mono` part. 

If any particular project really needs json-api, it will go only for that project.

Reply:
@fjgarlin could you please share some projects where json-api worked well? I'm curious to see how it is used there.

Reply:
Hey @Leksat, for one of our clients, they just needed to export certain area of the site (shop related) via an API so that data would be consumed by another team. They didn't have any specific needs on endpoints nor format of the data, so we just enabled the json-api modules and configured/exported everything.

Also, thinking on the cases where data will be consumed by other systems (not controlled by us), where those systems will just need endpoints and expected data, as opposed to needing to build a query/mutation.

My point from the beginning was (and is) to not prohibit it as it can have suitable use cases. I'm happy with GraphQL being the preferred option.